src/Arcus.Messaging.ServiceBus.Core/Extensions/ObjectExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Text;
+using Arcus.Messaging.Abstractions;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Newtonsoft.Json;
+
+namespace Arcus.Messaging.ServiceBus.Core.Extensions
+{
+    public static class MessageExtensions
+    {
+        private const string JsonContentType = "application/json";
+
+        /// <summary>
+        ///     Creates an Azure Service Bus Message for a message body
+        /// </summary>
+        /// <param name="messageBody">Body of the Service Bus message to process</param>
+        /// <param name="operationId">Unique identifier that spans one operation end-to-end</param>
+        /// <param name="transactionId">Unique identifier that spans one or more operations and are considered a transaction/session</param>
+        /// <param name="encoding">Encoding to use during serialization. Defaults to UTF8</param>
+        /// <returns>Azure Service Bus Message</returns>
+        public static Message WrapInServiceBusMessage(this object messageBody, string operationId = null, string transactionId = null, Encoding encoding = null)
+        {
+            Guard.NotNull(messageBody, nameof(messageBody));
+
+            encoding = encoding ?? Encoding.UTF8;
+
+            string serializedMessageBody = JsonConvert.SerializeObject(messageBody);
+            byte[] rawMessage = encoding.GetBytes(serializedMessageBody);
+
+            var serviceBusMessage = new Message(rawMessage)
+            {
+                UserProperties =
+                {
+                    { PropertyNames.ContentType, JsonContentType },
+                    { PropertyNames.Encoding, encoding.WebName }
+                },
+                CorrelationId = operationId
+            };
+
+            if (string.IsNullOrWhiteSpace(transactionId) == false)
```

Comment:
So if we receive a message, and we get the `TransactionID = ""` property, we consider this as a valid ID, but if we wrap the message body, we _don't_ consider this a valid ID?
Some ambigious representation?

Reply:
We're not saying that it's not valid, we're just saying that it doesn't matter as it's empty anyway.

Ignoring is different from blocking, or what are you referring to with "we consider this as a valid ID"?

Reply:
> Ignoring is different from blocking, or what are you referring to with "we consider this as a valid ID"?

In the `ObjectExtensions` we ignore the transaction ID if blank, so I would consider this means that there is not transaction ID defined; meaning 'blank' is not a valid transaction ID.
But when we retrieve the transaction ID from the message (`MessageExtensions`) and place it in the `MessagingContext`; we set the `.TransactionId` property with either a blank or non-blank value; meaning here that the transaction ID _can_ be blank and considered valid.

Just a bit a double interpretation for me.

Reply:
If it's empty than it's blank, so there is no point in assigning it. Don't really see the point to assign it?

Reply:
So, an empty or blank transaction ID _isn't_ considered valid? Then the `MessagePump.cs` should probably be changed to use `null` or an optional type when assigining the `MessagingContext.TransactionId` after calling `.GetTransactionId` on the message?

Reply:
An empty or blank transaction is valid, we just ignore it as it doesn't bring any value. And with valid I mean that we'll take it and ignore it, but won't throw an exception for it.

The pump itself should not make any logic on that imo, everything is determined by the publisher, not the consumer. If it's blank/empty/specified, then that's what we'll use.

Reply:
Ok, makes sense. So the `MessagePump.cs` and `ObjectExtensions.cs` may use something else, but each for a different reason.
Ok, then I'll resolve this.